src/components/dimensions/DimensionItemsSelect.js

```diff
@@ -1,57 +1,65 @@
+import { useDataQuery } from '@dhis2/app-runtime'
 import i18n from '@dhis2/d2-i18n'
 import PropTypes from 'prop-types'
 import React, { useEffect } from 'react'
-import { connect } from 'react-redux'
-import { loadDimensionItems } from '../../actions/dimensions.js'
 import { SelectField } from '../core/index.js'
+import { useUserSettings } from '../UserSettingsProvider.js'
 import styles from './styles/DimensionItemsSelect.module.css'
 
-const DimensionItemsSelect = ({
-    dimension,
-    items,
-    value,
-    onChange,
-    loadDimensionItems,
-}) => {
-    useEffect(() => {
-        if (dimension && !items) {
-            loadDimensionItems(dimension)
+// Load dimension items
+const DIMENSION_ITEMS_QUERY = {
+    items: {
+        resource: 'dimensions',
+        id: ({ id }) => `${id}/items`,
+        params: ({ nameProperty }) => ({
+            fields: ['id', `${nameProperty}~rename(name)`],
+            order: 'displayName:asc',
```

Comment:
Will `order` work when it's always `displayName`?

Reply:
Changed to `${nameProperty}:asc`